davidpak/SmartNote
# Added Server Config and Better Command Line Processing

## What type of PR is this? (check all applicable)


## Description

Added support for `config.json` in the working directory in which the server is run. The file is parsed if it exists which sets up how the server should run. If it doesn't exist, it is created.

Added utility classes for parsing arguments from the command line.

## QA Instructions, Screenshots, Recordings

To generate a `config.json` file, run the server once, and it will fill it with the default values. The file may be edited so that when you run the server, you do not have to modify everything on the command line. Specifying command line arguments overrides the values in the config file, but does not modify it.

Use `ant run -Dargs="--help"` to see all command line options.

To specify additional command line arguments and configurations, consult the classes in `cli` package. The files `Config.java`, `ServerConfig.java` and `ResourceConfig.java` make use of these utilities. If you add an option, be sure to update the help message in `Server.printHelp()`.

## Added/updated tests?
